scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,57 +3,348 @@
-
-        elseif
-            sLux and
-            sLuna and
-            sAstrum and
-            count == 3 and
-            (os.time() <= lastCrimson or hasCrimsonKey)
-        then
-            player:messageSpecial(ID.text.DRYEYES_3, xi.ki.CRIMSON_KEY)
-        -- White Coral Key:
-        -- elseif xi.settings.main.ENABLE_AMK == 1 and
-            -- haven't even started AMK related trades yet.
+        return true
+    end
+end
+
+local function takeReqKeyItems(player, ki)
+    local entry = menuMetadata[1][ki]
+    for i, reqKeyItem in pairs(entry.reqKeyItems) do
+        player:delKeyItem(reqKeyItem)
+        player:messageSpecial(ID.text.KEYITEM_LOST, reqKeyItem)
+    end
+
+    if entry.prereqKeyItems then
+        for i, reqKeyItem in pairs(entry.prereqKeyItems) do
+            player:delKeyItem(reqKeyItem)
+            player:messageSpecial(ID.text.KEYITEM_LOST, reqKeyItem)
+        end
+    end
+
+    player:addKeyItem(ki)
+    player:setCharVar(entry.charVar, getMidnight())
+    player:showText(player, ID.text.DRYEYES_2)
+    player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+end
+
+entity.onTrade = function(player, npc, trade)
+    local count  = trade:getItemCount()
+    local ki     = xi.ki.MOOGLE_KEY
+    local asaKit = player:getCharVar(menuMetadata[1][ki].reqItemCharVar)
+
+    if -- Crimson Key: Trade Seedspall's Lux, Luna, Astrum
+        trade:hasItemQty(menuMetadata[1][xi.ki.CRIMSON_KEY].reqItems[1], 1)   and
+        trade:hasItemQty(menuMetadata[1][xi.ki.CRIMSON_KEY].reqItems[2], 1)   and
+        trade:hasItemQty(menuMetadata[1][xi.ki.CRIMSON_KEY].reqItems[3], 1)   and
+        count == 3                                                            and
+        player:getCurrentMission(xi.mission.log_id.ACP) == xi.mission.id.acp.A_CRYSTALLINE_PROPHECY_FIN
+    then
+        ki = xi.ki.CRIMSON_KEY
+        tradeForKeyItem(player, trade, ki)
+    elseif -- White Coral Key: orcish plate armor, quadav backscale, yagudo caulk
+        trade:hasItemQty(menuMetadata[1][xi.ki.WHITE_CORAL_KEY].reqItems[1], 1)   and
+        trade:hasItemQty(menuMetadata[1][xi.ki.WHITE_CORAL_KEY].reqItems[2], 1)   and
+        trade:hasItemQty(menuMetadata[1][xi.ki.WHITE_CORAL_KEY].reqItems[3], 1)   and
+        count == 3                                                                and
+        player:getCurrentMission(xi.mission.log_id.AMK) == xi.mission.id.amk.A_MOOGLE_KUPO_DETAT_FIN
+    then
+        ki = xi.ki.WHITE_CORAL_KEY
+        tradeForKeyItem(player, trade, ki)
+    elseif -- Moogle Key: trade proper enfeebling kit
+        asaKit ~= 0                                                    and
+        trade:hasItemQty(asaKit, 1)                                    and
+        count == 1                                                     and
+        player:getCurrentMission(xi.mission.log_id.ASA) == xi.mission.id.asa.A_SHANTOTTO_ASCENSION_FIN
+    then
```

Comment:
The arbitrary gaps between the statement and the and/or keywords aren't needed here. Only put in extraneous whitespace if you're aligning things to something: like variable names or the = operators

Reply:
Ok so just

```
Statement and
Longer statement and
Short and
Blah
```
?

Reply:
Yessir